### PR TITLE
Relax type-constraint for `Transition` to allow `AbstractArray`

### PR DESCRIPTION
**Pull request by torfjelde**
It does what it says in the title, my amigos :heart:

### CODE REVIEW

File: src/AdvancedMH.jl [hunk 1]
```diff
@@ -50,7 +50,7 @@ end
 
 # Create a very basic Transition type, only stores the
 # parameter draws and the log probability of the draw.
-struct Transition{T<:Union{Vector, Real, NamedTuple}, L<:Real} <: AbstractTransition
+struct Transition{T<:Union{AbstractVector, Real, NamedTuple}, L<:Real} <: AbstractTransition
```
Comment (devmotion) [inline src/AdvancedMH.jl:53]: Is the type constraint useful? Or should we just define
```suggestion
struct Transition{T,L<:Real} <: AbstractTransition
```
?

Comment (torfjelde) [reply src/AdvancedMH.jl:53]: I thought the same thing just after making the suggestion. I'm very fine with removing it completely, but I'm not too familiar with internals of AMH.jl :shrug:

Comment (torfjelde) [reply src/AdvancedMH.jl:53]: So you / @cpfiffer make the call!

Comment (cpfiffer) [reply src/AdvancedMH.jl:53]: Hm. I don't mind either way, honestly, which suggests to me that we should just remove the type constraint entirely.

**Review by cpfiffer — APPROVED** [review]

**Comment by cpfiffer** [comment]
Could you bump the patch version?

**Comment by cpfiffer** [comment]
Let's goooooooo

**Pull request merged**